core/google/cloud/connection.py

```diff
@@ -273,7 +273,7 @@ def _do_request(self, method, url, headers, data,
                                  body=data)
 
     def api_request(self, method, path, query_params=None,
-                    data=None, content_type=None,
+                    data=None, content_type=None, headers=None,
```

Comment:
I'd say YAGNI:  except for `Blob.rewrite`, they get passed via the `google.cloud.streaming` APIs, rather than our normal JSON connection class.